### PR TITLE
Use kubectl to manage config for gce clusters

### PR DESCRIPTION
**Pull request by j3ffml**
kube-up.sh now saves files in a subdirectory of `~/.kube` and sets global kubeconfig so that turning up a gce cluster (including e2e) doesn't squash existing config.

/cc @filbranden

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -505,30 +505,44 @@ function kube-up {
...
   # TODO: generate ADMIN (and KUBELET) tokens and put those in the master's
   # config file.  Distribute the same way the htpasswd is done.
-  (umask 077
-   gcloud compute ssh --project "${PROJECT}" --zone "$ZONE" "${MASTER_NAME}" --command "sudo cat /srv/kubernetes/kubecfg.crt" >"${HOME}/${kube_cert}" 2>/dev/null
-   gcloud compute ssh --project "${PROJECT}" --zone "$ZONE" "${MASTER_NAME}" --command "sudo cat /srv/kubernetes/kubecfg.key" >"${HOME}/${kube_key}" 2>/dev/null
-   gcloud compute ssh --project "${PROJECT}" --zone "$ZONE" "${MASTER_NAME}" --command "sudo cat /srv/kubernetes/ca.crt" >"${HOME}/${ca_cert}" 2>/dev/null
-
-   cat << EOF > ~/.kubernetes_auth
+  (
+   mkdir -p $"${config_dir}"
```
Comment (filbranden) [inline cluster/gce/util.sh]: This is incorrect, `$"..."` means internationalize the string, you need to remove the `$` that comes right before the double quote here.

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -505,30 +505,44 @@ function kube-up {
...
+   gcloud compute ssh --project "${PROJECT}" --zone "$ZONE" "${MASTER_NAME}" --command "sudo cat /srv/kubernetes/kubecfg.crt" >"${config_dir}/${kube_cert}" 2>/dev/null
+   gcloud compute ssh --project "${PROJECT}" --zone "$ZONE" "${MASTER_NAME}" --command "sudo cat /srv/kubernetes/kubecfg.key" >"${config_dir}/${kube_key}" 2>/dev/null
+   gcloud compute ssh --project "${PROJECT}" --zone "$ZONE" "${MASTER_NAME}" --command "sudo cat /srv/kubernetes/ca.crt" >"${config_dir}/${ca_cert}" 2>/dev/null
+
+   "${kubectl}" config set-cluster "${context}" --server="https://${KUBE_MASTER_IP}" --certificate-authority="${config_dir}/${ca_cert}" --global
+   "${kubectl}" config set-credentials "${user}" --auth-path="${config_dir}/${kube_auth}" --global
+   "${kubectl}" config set-context "${context}" --cluster="${context}" --user="${user}" --global
+   "${kubectl}" config use-context "${context}" --global
+
+   cat << EOF > "${config_dir}/$kube_auth"
```
Comment (filbranden) [inline cluster/gce/util.sh]: Nit: curly braces around variable name `${config_dir}/${kube_auth}`

File: cluster/gce/util.sh [hunk 3]
```diff
@@ -505,30 +505,44 @@ function kube-up {
...
+  "KeyFile": "${config_dir}/${kube_key}"
 }
 EOF
 
-   chmod 0600 ~/.kubernetes_auth "${HOME}/${kube_cert}" \
-     "${HOME}/${kube_key}" "${HOME}/${ca_cert}"
-   echo Wrote ~/.kubernetes_auth
+   chmod 0600 "${config_dir}/${kube_auth}" "${config_dir}/$kube_cert" \
+     "${config_dir}/${kube_key}" "${config_dir}/${ca_cert}"
+   echo "Wrote ${config_dir}/$kube_auth"
```
Comment (filbranden) [inline cluster/gce/util.sh]: Nit: curly braces around variable name `${config_dir}/${kube_auth}`

File: cluster/gce/util.sh [hunk 4]
```diff
@@ -576,7 +590,7 @@ EOF
   echo
   echo -e "${color_yellow}  https://${KUBE_MASTER_IP}"
   echo
-  echo -e "${color_green}The user name and password to use is located in ~/.kubernetes_auth.${color_norm}"
+  echo -e "${color_green}The user name and password to use is located in ${config_dir}/$kube_auth.${color_norm}"
```
Comment (filbranden) [inline cluster/gce/util.sh]: Nit: curly braces around variable name `${config_dir}/${kube_auth}`

**Comment by filbranden** [comment]
Looks alright. I'm not 100% sure I grasp all the implications of this directory change... I imagine this change was discussed with others who support the move to a subdirectory?

File: cluster/gce/util.sh (hunk 1 above)
Comment (jlowdermilk) [inline cluster/gce/util.sh]: Done.

File: cluster/gce/util.sh (hunk 2 above)
Comment (jlowdermilk) [inline cluster/gce/util.sh]: Done.

File: cluster/gce/util.sh (hunk 3 above)
Comment (jlowdermilk) [inline cluster/gce/util.sh]: Done.

File: cluster/gce/util.sh (hunk 4 above)
Comment (jlowdermilk) [inline cluster/gce/util.sh]: Done.

**Comment by jlowdermilk** [comment]
Ping. I'd like to merge this today.

**Comment by brendandburns** [comment]
LGTM.

**Pull request merged**